src/main/java/duke/parser/Parser.java

```diff
@@ -0,0 +1,127 @@
+                    String unmarkedTask = String.format(" %s", tsk);
+                    String[] unmarkedMsg = {markMsg, unmarkedTask};
+                    this.ui.printMsg(unmarkedMsg);
+                    break;
+                case "todo":
+                    if (inputs.length == 1)
+                        throw new DukeException("☹ OOPS!!! The description of a todo cannot be empty.");
+                    tsk = new Todo(inputs[1]);
+                    data.add(tsk);
+                    tskNum = String.format("Now you have %d tasks in the list.", data.size());
+                    String[] todoMsg = {cfmMsg, " " + tsk.toString(), tskNum};
+                    this.ui.printMsg(todoMsg);
+                    break;
+                case "deadline":
+                    desc = inputs[1].split(" /by ");
+                    date = parseDate(desc[1]);
+                    tsk = new Deadline(desc[0], date);
+                    data.add(tsk);
+                    tskNum = String.format("Now you have %d tasks in the list.", data.size());
+                    String[] deadlineMsg = {cfmMsg, " " + tsk.toString(), tskNum};
+                    this.ui.printMsg(deadlineMsg);
+                    break;
+                case "event":
+                    desc = inputs[1].split(" /from | /to ");
+                    from = parseDate(desc[1]);
+                    to = parseDate(desc[2]);
+                    tsk = new Event(desc[0], from, to);
+                    data.add(tsk);
+                    tskNum = String.format("Now you have %d tasks in the list.", data.size());
+                    String[] eventMsg = {cfmMsg, " " + tsk.toString(), tskNum};
+                    this.ui.printMsg(eventMsg);
+                    break;
+                case "delete":
+                    idx = Integer.parseInt(inputs[1]) - 1;
+                    tsk = data.get(idx);
+                    data.delete(idx);
+                    tskNum = String.format("Now you have %d tasks in the list.", data.size());
+                    String[] delMsg = {"Noted. I've removed this duke.exceptions.task:", tsk.toString(), tskNum};
+                    this.ui.printMsg(delMsg);
+                    break;
+                default:
+                    throw new DukeException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+
+            }
+            this.storage.saveTasks(data);
+        } catch (DukeException | IllegalArgumentException e) {
+            this.ui.printMsg(e.getMessage());
+        }
+        return true;
+    }
+
+    private LocalDateTime parseDate(String dateString) {
+        LocalDateTime date;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("M/d/yyyy HHmm");
+        try {
+            date = LocalDateTime.parse(dateString, formatter);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format: " + dateString + ". Please give your date" +
```

Comment:
Throwing this exception does not help to identify the specific input error. Is there any way to prevent this?

Reply:
Correct date format will be displayed to the user so that they know that the date format of the input is wrong.